front50-gcs/src/main/java/com/netflix/spinnaker/front50/model/GcsStorageService.java

```diff
@@ -512,33 +512,33 @@ private AtomicBoolean scheduledUpdateLock(String daoTypeName) {
 
   @VisibleForTesting
   public void scheduleWriteLastModified(String daoTypeName) {
+    Date when = new Date();
+    when.setSeconds(when.getSeconds() + 2);
+    GcsStorageService service = this;
+    Runnable task = new Runnable() {
+      public void run() {
+        // Release the scheduled update lock, and perform the actual update
+        scheduledUpdateLock(daoTypeName).set(false);
+        log.info("RUNNING {}", daoTypeName);
+        service.writeLastModified(daoTypeName);
+      }
+    };
     if (scheduledUpdateLock(daoTypeName).compareAndSet(false, true)) {
-      Date when = new Date();
-      GcsStorageService service = this;
-      Runnable task = new Runnable() {
-        public void run() {
-          log.info("RUNNING {}", daoTypeName);
-          // Release the scheduled update lock, and perform the actual update
-          scheduledUpdateLock(daoTypeName).set(false);
-          service.writeLastModified(daoTypeName);
-        }
-      };
-      when.setSeconds(when.getSeconds() + 2);
       log.info("Scheduling deferred update {} timestamp.", daoTypeName);
       taskScheduler.schedule(task, when);
     }
   }
 
   private void writeLastModified(String daoTypeName) {
+    // We'll just touch the file since the StorageObject manages a timestamp.
```

Comment:
I dont follow why you moved this outside the guard.

Reply:
This was to reduce the amount of code running between acquiring the lock and releasing it, as we'd end up in a bad state if an exception was thrown between acquiring and releasing the lock.  I could have wrapped this in a `try` and released the lock in a `catch` but it seemed cleaner to just move it before acquiring the lock at all...but open to doing the `catch` if you think that's cleaner.

Reply:
you can leave it as is.